hikari/utilities/iterators.py

```diff
@@ -497,19 +516,32 @@ async def last(self) -> ValueT:
 
         Returns
         -------
-        T
+        ValueT
             The last result.
 
+        !!! note
+            This method will deplete the iterator if run.
```

Comment:
Maybe `advance` is a better word than `deplete`?

I'm also wondering if we can provide a `peek` method that does the same thing but does not displace the item from the iterator.

E.g.
```py
# Use a dummy enum as we can use `typing.Literal` with these!
class _PeekSentinel(enum.Enum):
    NO_VALUE = enum.auto()

_peek: Union[ValueT, Literal[_PeekSeninel.NO_VALUE]] 

async def peek(self) -> ValueT:
    if self._peek is self._PeekSentinel.NO_VALUE:
        item = await self.next()
        self._peek = item
    return self._peek

async def __anext__(self) -> ValueT:
    if self._peek is not self._PeekSentinel.NO_VALUE:
        item = self._peek
        self._peek = self._PeekSentinel.NO_VALUE
        return item
    else:
        # existing code.
```

This would just mean that peeking would cache the next value eagerly and then clear it next time __anext__ is called.

Reply:
"advance" doesn't really communicate the fact that it totally uses up the lazy iterator

Reply:
it is an iterator. By definition a python iterator gets "used up" by advancing.

Consume is probably what I meant, rather than advance, come to think of it.